cocotb/share/include/gpi.h

```diff
@@ -97,11 +107,13 @@ typedef struct gpi_sim_info_s
     int32_t   *reserved[4];
 } gpi_sim_info_t;
 
+
+
 // Define a type for our simulation handle.
-typedef void * gpi_sim_hdl;
+typedef struct GpiObjHdl *gpi_sim_hdl;
 
 // Define a handle type for iterators
-typedef void * gpi_iterator_hdl;
+typedef struct GpiIterator *gpi_iterator_hdl;
```

Comment:
By doing this, it now becomes a compilation error to pass one in place of the other.